sdk/storage/azure-storage-file/apireview/azure/storage/file/share_client.pyi

```diff
@@ -0,0 +1,39 @@
+from .models import AccessPolicy, SharePermissions, ShareProperties
+from typing import Any, Dict, Optional, Union
+
+class ShareClient(StorageAccountHostsMixin):
+    snapshot: Any = ...
+    share_name: Any = ...
+    def __init__(self, share_url: str, share: Optional[Union[str, ShareProperties]]=..., snapshot: Optional[Union[str, Dict[str, Any]]]=..., credential: Optional[Any]=..., **kwargs: Any) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, share: Union[str, ShareProperties], snapshot: Optional[str]=..., credential: Optional[Any]=..., **kwargs: Any) -> ShareClient: ...
+    def generate_shared_access_signature(self, permission: Optional[Union[SharePermissions, str]]=..., expiry: Optional[Union[datetime, str]]=..., start: Optional[Union[datetime, str]]=..., policy_id: Optional[str]=..., ip: Optional[str]=..., protocol: Optional[str]=..., cache_control: Optional[str]=..., content_disposition: Optional[str]=..., content_encoding: Optional[str]=..., content_language: Optional[str]=..., content_type: Any=...) -> str: ...
+    def get_directory_client(self, directory_path: Optional[str]=...) -> DirectoryClient: ...
+    def get_file_client(self, file_path: str) -> FileClient: ...
+    def create_share(self, metadata: Optional[Dict[str, str]]=..., quota: Optional[int]=..., timeout: Optional[int]=..., **kwargs: Optional[Any]) -> Dict[str, Any]: ...
+    def create_snapshot(self, metadata: Optional[Dict[str, str]]=..., timeout: Optional[int]=..., **kwargs: Optional[Any]) -> Dict[str, Any]: ...
+    def delete_share(self, delete_snapshots: Optional[bool]=..., timeout: Optional[int]=..., **kwargs: Any) -> None: ...
+    def get_share_properties(self, timeout: Optional[int]=..., **kwargs: Any) -> ShareProperties: ...
+    def set_share_quota(self, quota: int, timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def set_share_metadata(self, metadata: Dict[str, Any], timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def get_share_access_policy(self, timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def set_share_access_policy(self, signed_identifiers: Optional[Dict[str, Optional[AccessPolicy]]]=..., timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, str]: ...
+    def get_share_stats(self, timeout: Optional[int]=..., **kwargs: Any) -> int: ...
+    def list_directories_and_files(self, directory_name: Optional[str]=..., name_starts_with: Optional[str]=..., marker: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> Iterable[dict[str, str]]: ...
+    def create_permission_for_share(self, file_permission: str, timeout: Optional[int]=..., **kwargs: Any) -> str: ...
+    def get_permission_for_share(self, file_permission_key: str, timeout: Optional[int]=..., **kwargs: Any) -> str: ...
+    def create_directory(self, directory_name: str, metadata: Optional[Dict[str, Any]]=..., timeout: Optional[int]=..., **kwargs: Any) -> DirectoryClient: ...
```

Comment:
We have `create_directory` but don't seem to have `delete_directory`?

Reply:
Please add :)

Reply:
#7835